math/mathcore/inc/Fit/EvaluateLogL.hxx

```diff
@@ -0,0 +1,569 @@
+               for (unsigned int parameterIndex = 0; parameterIndex < npar; parameterIndex++)
+                  result[parameterIndex] += pointContributionVec[parameterIndex];
+            }
+
+            return result;
+         };
+
+         std::vector<T> gVec(npar);
+         std::vector<double> g(npar);
+
+#ifndef R__USE_IMT
+         // to fix compiler warning
+         (void)nChunks;
+
+         // If IMT is disabled, force the execution policy to the serial case
+         if (executionPolicy == ROOT::Fit::ExecutionPolicy::kMultithread) {
+            Warning("FitUtil::EvaluateLogLGradient",
+                    "Multithread execution policy requires IMT, which is disabled. Changing "
+                    "to ROOT::Fit::ExecutionPolicy::kSerial.");
+            executionPolicy = ROOT::Fit::ExecutionPolicy::kSerial;
+         }
+#endif
+
+         if (executionPolicy == ROOT::Fit::ExecutionPolicy::kSerial) {
+            ROOT::TSequentialExecutor pool;
+            gVec = pool.MapReduce(mapFunction, ROOT::TSeq<unsigned>(0, numVectors), redFunction);
+         }
+#ifdef R__USE_IMT
+         else if (executionPolicy == ROOT::Fit::ExecutionPolicy::kMultithread) {
+            ROOT::TThreadExecutor pool;
+            auto chunks = nChunks != 0 ? nChunks : setAutomaticChunking(numVectors);
+            gVec = pool.MapReduce(mapFunction, ROOT::TSeq<unsigned>(0, numVectors), redFunction, chunks);
+         }
+#endif
+         else {
+            Error("FitUtil::EvaluateLogLGradient", "Execution policy unknown. Avalaible choices:\n "
+                                                   "ROOT::Fit::ExecutionPolicy::kSerial (default)\n "
+                                                   "ROOT::Fit::ExecutionPolicy::kMultithread (requires IMT)\n");
+         }
+
+         // Compute the contribution from the remaining points
+         unsigned int remainingPoints = initialNPoints % vecSize;
+         if (remainingPoints > 0) {
+            auto remainingPointsContribution = mapFunction(numVectors);
+            // Add the contribution from the valid remaining points and store the result in the output variable
+            auto remainingMask = ::vecCore::Int2Mask<T>(initialNPoints % vecSize);
+            for (unsigned int param = 0; param < npar; param++) {
+               vecCore::MaskedAssign(gVec[param], remainingMask, gVec[param] + remainingPointsContribution[param]);
+            }
+         }
+         // reduce final gradient result from T to double
+         for (unsigned int param = 0; param < npar; param++) {
+            grad[param] = vecCore::ReduceAdd(gVec[param]);
+         }
+
+#ifdef DEBUG_FITUTIL
+         std::cout << "Final gradient ";
+         for (unsigned int param = 0; param < npar; param++) {
+            std::cout << "  " << grad[param];
+         }
+         std::cout << "\n";
+#endif
+      }
+   };
+
+   template<>
+   struct LogL<double>{
+#endif
```

Comment:
This is another strange `#endif` that seems to break things if the code above it is part of the header.